x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/InferenceContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.inference;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Record for storing context alongside an inference request, typically used for metadata.
+ * This is mainly used to pass along inference context on the transport layer without relying on {@link org.elasticsearch.common.util.concurrent.ThreadContext},
+ * which depending on the internal {@link org.elasticsearch.client.internal.Client} throws away parts of the context, when passed along the transport layer.
+ *
+ * @param productUseCase - for now mainly interesting to Elastic Inference Service
+ */
+public record InferenceContext(String productUseCase) implements Writeable, ToXContent {
+
+    public InferenceContext(StreamInput in) throws IOException {
+        this(in.readString());
+    }
+
+    public static InferenceContext empty() {
```

Comment:
How about we create a static instance that way we don't create multiple empty ones? Something like this:

```
public static final InferenceContext EMPTY_INSTANCE = new InferenceContext("");
```

Reply:
Good point! Adjusted with [Replace InferenceContext.empty() with InferenceContext.EMPTY_INSTANCE](https://github.com/elastic/elasticsearch/pull/124025/commits/1c0a3c476cc8697fe695a0f2def381b12af593a9)